grpcbp/server_middlewares_test.go

```diff
@@ -181,3 +184,123 @@ func (t *mockService) PingList(req *pb.PingRequest, c pb.TestService_PingListSer
 func (t *mockService) PingStream(c pb.TestService_PingStreamServer) error {
 	panic("not implemented")
 }
+
+func TestInjectPrometheusUnaryServerClientInterceptor(t *testing.T) {
+	const (
+		serviceName = "mwitkow.testproto.TestService"
+		serverSlug  = "example-preference-server"
+		method      = "Ping"
+	)
+	// create test server with InjectPrometheusUnaryServerInterceptor
+	l, service := setupServer(t, grpc.UnaryInterceptor(
+		InjectPrometheusUnaryServerInterceptor(),
+	))
+
+	// create test client
+	conn := setupClient(t, l, grpc.WithUnaryInterceptor(
+		PrometheusUnaryClientInterceptor(serverSlug),
+	))
+
+	// instantiate gRPC client
+	client := pb.NewTestServiceClient(conn)
+
+	testCases := []struct {
+		name    string
+		wantErr codes.Code
+		code    string
+		success string
+		method  string
+	}{
+		{
+			name:    "success",
+			wantErr: codes.OK,
+			code:    "OK",
+			success: "true",
+			method:  "Ping",
+		},
+		{
+			name:    "err",
+			wantErr: codes.Internal,
+			code:    "Unknown",
+			success: "false",
+			method:  "PingError",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			serverLatencyDistribution.Reset()
+			serverRPCRequestCounter.Reset()
+			serverActiveRequests.Reset()
+			clientLatencyDistribution.Reset()
+			clientRPCRequestCounter.Reset()
+			clientActiveRequests.Reset()
+
+			serverLatencyLabels := prometheus.Labels{
+				serviceLabel: serviceName,
+				methodLabel:  tt.method,
+				typeLabel:    unary,
+				successLabel: tt.success,
+			}
+
+			serverTotalRequestLabels := prometheus.Labels{
+				serviceLabel: serviceName,
+				methodLabel:  tt.method,
+				typeLabel:    unary,
+				successLabel: tt.success,
+				codeLabel:    tt.code,
+			}
+
+			serverActiveRequestLabels := prometheus.Labels{
+				serviceLabel: serviceName,
+				methodLabel:  tt.method,
+			}
+
+			clientLatencyLabels := prometheus.Labels{
+				serviceLabel:    serviceName,
+				methodLabel:     tt.method,
+				typeLabel:       unary,
+				successLabel:    tt.success,
+				serverSlugLabel: serverSlug,
+			}
+
+			clientTotalRequestLabels := prometheus.Labels{
+				serviceLabel:    serviceName,
+				methodLabel:     tt.method,
+				typeLabel:       unary,
+				successLabel:    tt.success,
+				serverSlugLabel: serverSlug,
+				codeLabel:       tt.code,
+			}
+
+			clientActiveRequestLabels := prometheus.Labels{
+				serviceLabel:    serviceName,
+				methodLabel:     tt.method,
+				serverSlugLabel: serverSlug,
+			}
+
+			defer promtest.NewPrometheusMetricTest(t, "server latency", serverLatencyDistribution, serverLatencyLabels).CheckExists()
+			defer promtest.NewPrometheusMetricTest(t, "server rpc count", serverRPCRequestCounter, serverTotalRequestLabels).CheckDelta(1)
```

Comment:
Some of these are named almost identically, e.g. "clientActiveRequests" <-> "clientActiveRequestLabels" -- maybe the rest should be unified too?  This (RPC vs Total) stands out the most